python/pyspark/worker.py

```diff
@@ -500,7 +501,10 @@ def main(infile, outfile):
 
             except (resource.error, OSError, ValueError) as e:
                 # not all systems support resource limits, so warn instead of failing
-                print("WARN: Failed to set memory limit: {0}\n".format(e), file=sys.stderr)
+                warnings.warn(
+                    "Failed to set memory limit: {0}\n".format(e),
+                    ResourceWarning
+                )
```

Comment:
I'm not sure we can change this to warnings. Where will the message go?

Reply:
They are sent to stderr as currently, but yeah, I don't believe they will end up in a log by default for a remote worker.

There are also similar print statements above this one, what about those?

Reply:
oh, yeah. let's revert this one.

Reply:
Good catch. Could use `warnings.showwarning` instead? It supports writing to `_io.TextIOWrapper`, so we could probably do something similar to this:

```python
warnings.showwarning(
    "Failed to set memory limit: {0}\n".format(e),
    ResourceWarning,
    __file__,
    0,  # Getting line number without inspect will be tricky
    sys.stdout
)
```

Reply:
That sounds fine if it works though we will have to manually test it out. I am fine either way (reverting it or fixing as avoid after manually testing). We could also use [formatwarning](https://docs.python.org/3/library/warnings.html#warnings.formatwarning) that will also obviously work. Maybe you could get the file number via, for example, `inspect.getframeinfo(inspect.currentframe()).lineno` too.

I am okay either way if it works.

Reply:
> Maybe you could get the file number via, for example, `inspect.getframeinfo(inspect.currentframe()).lineno` too.

My concern is that it is CPython-specific solution (I looked into this before, when trying to find viable replacement for keyword capturing mechanism in `keyword_only`). Since the only other implementation we support is PyPy it is probably fine though.